pkg/plugins/document/document.go

```diff
@@ -35,17 +33,16 @@ var validOperators = map[string]bool{
 // ValidateKey - validates a document key, used for operations on a single document e.g. Get, Set, Delete
 func ValidateKey(key *Key) error {
 	if key == nil {
-		return plugins.NewInvalidArgError("provide non-nil key")
```

Comment:
Pre-emptive comment on removing these:

As the plugin standard errors themselves, are returned and translated directly into grpc errors by the adapters, nesting these errors in shared functionality starts to become confusing as it becomes unclear what errors are ok to directly pass though and what errors need to be wrapped for correct wire translation (without looking at the underlying code every-time). And re-wrapping already wrapped errors starts to produce vestigial/redundant information and only the top most error code is translated by the adapters.

I've adopted a policy here that internal or shared methods will produce simple and instructive error messages, and the plugin methods will directly return the wrapped `PluginError`.

Let me know of any thoughts on this.